File: benchmarks/compare/lowlevel_arena.py
class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

def compute(n):
    total = 0
    i = 0
    while i < n:
        p = Point(i, i + 1)
        nums = [1, 2, 3, 4, 5, 6, 7, 8]
        j = 0
        inner = 0
        while j < 8:
            inner = inner + nums[j]
            j = j + 1
        total = total + p.x + p.y + inner
        i = i + 1
    return total

print(compute(5000000))

File: benchmarks/compare/lowlevel_arena.c
#include <stdio.h>
long long compute(long long n) {
    long long total = 0;
    long long i = 0;
    long long nums[8] = {1,2,3,4,5,6,7,8};
    while (i < n) {
        long long x = i, y = i + 1;
        long long inner = 0;
        for (int j = 0; j < 8; j++) inner += nums[j];
        total += x + y + inner;
        i++;
    }
    return total;
}
int main(void) {
    printf("%lld\n", compute(5000000));
    return 0;
}
